tokio-executor/src/enter.rs

```diff
@@ -67,6 +67,30 @@ pub fn enter() -> Result<Enter, EnterError> {
     })
 }
 
+// Forces the current "entered" state to be cleared while the closure
+// is executed.
+//
+// # Warning
+//
+// This is hidden for a reason. Do not use without fully understanding
+// executors. Misuing can easily cause your program to deadlock.
+#[doc(hidden)]
+pub fn exit<F: FnOnce() -> R, R>(f: F) -> R {
+    ENTERED.with(|c| {
+        debug_assert!(c.get());
+        c.set(false);
+    });
+
+    let ret = f();
+
+    ENTERED.with(|c| {
```

Comment:
Do you think we need a drop guard to reset? Types could expect the executor available in drop fns

Reply:
You mean in case the closure panics?